opentelemetry-exporter-geneva/geneva-uploader/src/client.rs

```diff
@@ -0,0 +1,107 @@
+//! High-level GenevaClient for user code. Wraps config_service and ingestion_service.
+
+use crate::config_service::client::{AuthMethod, GenevaConfigClient, GenevaConfigClientConfig};
+use crate::ingestion_service::uploader::{GenevaUploader, GenevaUploaderConfig};
+use crate::payload_encoder::lz4_chunked_compression::lz4_chunked_compression;
+use crate::payload_encoder::otlp_encoder::OtlpEncoder;
+use opentelemetry_proto::tonic::logs::v1::ResourceLogs;
+use std::sync::Arc;
+
+/// Configuration for GenevaClient (user-facing)
+#[derive(Clone, Debug)]
+pub struct GenevaClientConfig {
+    pub endpoint: String,
+    pub environment: String,
+    pub account: String,
+    pub namespace: String,
+    pub region: String,
+    pub config_major_version: u32,
+    pub auth_method: AuthMethod,
+    pub tenant: String,
+    pub role_name: String,
+    pub role_instance: String,
+    // Add event name/version here if constant, or per-upload if you want them per call.
+}
+
+/// Main user-facing client for Geneva ingestion.
+#[derive(Clone)]
+pub struct GenevaClient {
+    uploader: Arc<GenevaUploader>,
```

Comment:
Do we want different instances of `GenevaClient` to share the uploader and encoder?